### ISSUE TITLE
Cannot find module 'weak' on mac?

### ISSUE THREAD

**Issue by reggi** [sgentle/phantomjs-node#307]
Not using windows and it's asking for dep `weak`.

```
thomas@workstation:shopify-endeavor$ npm run install
npm info it worked if it ends with ok
npm info using npm@2.13.3
npm info using node@v3.2.0
npm info preinstall shopify-endeavor@0.0.0
npm info install shopify-endeavor@0.0.0

> shopify-endeavor@0.0.0 install /Users/thomas/Desktop/shopify-endeavor
> foreman run -- node lib/shopify-install.js

module.js:338
    throw err;
    ^

Error: Cannot find module 'weak'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at new D (/Users/thomas/Desktop/shopify-endeavor/node_modules/phantom/node_modules/dnode/index.js:28:20)
    at module.exports (/Users/thomas/Desktop/shopify-endeavor/node_modules/phantom/node_modules/dnode/index.js:8:12)
    at /Users/thomas/Desktop/shopify-endeavor/node_modules/phantom/phantom.js:159:13
    at Server.handler (/Users/thomas/Desktop/shopify-endeavor/node_modules/phantom/node_modules/shoe/index.js:22:9)
    at emitOne (events.js:77:13)
    at Server.emit (events.js:169:7)
    at App.emit (/Users/thomas/Desktop/shopify-endeavor/node_modules/phantom/node_modules/shoe/node_modules/sockjs/lib/sockjs.js:182:27)
    at Session.emit_open (/Users/thomas/Desktop/shopify-endeavor/node_modules/phantom/node_modules/shoe/node_modules/sockjs/lib/transport.js:107:23)
    at doNTCallback0 (node.js:407:9)
    at process._tickCallback (node.js:336:13)

npm info shopify-endeavor@0.0.0 Failed to exec install script
npm ERR! Darwin 14.4.0
npm ERR! argv "/Users/thomas/.nvm/versions/io.js/v3.2.0/bin/iojs" "/Users/thomas/.nvm/versions/io.js/v3.2.0/bin/npm" "run" "install"
npm ERR! node v3.2.0
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE
npm ERR! shopify-endeavor@0.0.0 install: `foreman run -- node lib/shopify-install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the shopify-endeavor@0.0.0 install script 'foreman run -- node lib/shopify-install.js'.
npm ERR! This is most likely a problem with the shopify-endeavor package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     foreman run -- node lib/shopify-install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls shopify-endeavor
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/thomas/Desktop/shopify-endeavor/npm-debug.log
```

**Comment by vhchung** [comment]
You can try this: find dnode module in phantom folder
![dnode_package_json](https://cloud.githubusercontent.com/assets/7991775/9896009/a9913c62-5c60-11e5-97c3-41b2f0490610.png)
Then edit file package.json, change weak version from 0.4.1 to 1.0.0
![dnode_change](https://cloud.githubusercontent.com/assets/7991775/9896008/a990b814-5c60-11e5-910f-0d7b92e39779.png)
Then open Terminal in this folder, run `sudo npm i` to install weak.
This way solved the problem on my Mac and CentOS server too.

**Comment by donaldpipowitch** [comment]
This doesn't solve it for me on Linux. I think its this problem: https://github.com/substack/dnode/issues/169?

**Comment by AugustNagro** [comment]
I solved this by setting "dnode": ">=1.2.2", and this was recently [merged](https://github.com/sgentle/phantomjs-node/commit/336a67409dc5c0b179a4e3778f84e5952a205840)

Version 1.0.0 is still throwing errors on my MBP with node v4.1.0
